drivers/sensor/lis2de12/lis2de12.c

```diff
@@ -0,0 +1,479 @@
+/* ST Microelectronics LIS2DE12 3-axis accelerometer sensor driver
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2de12.pdf
+ */
+
+#define DT_DRV_COMPAT st_lis2de12
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "lis2de12.h"
+
+LOG_MODULE_REGISTER(LIS2DE12, CONFIG_SENSOR_LOG_LEVEL);
+
+static const float lis2de12_odr_map[10] = {
+			0.0f, 1.0f, 10.0f, 25.0f, 50.0f, 100.0f, 200.0f,
+			400.0f, 1620.0f, 5376.0f};
```

Comment:
sure you want float here? `lis2de12_freq_to_odr_val` takes freq as uint16_t and compares it with this, may be worth keep this on uint16_t as well

Reply:
Yes, it makes sense. It is already like these in the other stmemsc drivers. Thanks for catching it!